VideoLocker/src/main/java/org/edx/mobile/user/UserAPI.java

```diff
@@ -55,49 +52,116 @@
+                                          @NonNull final String username,
+                                          @NonNull final ErrorHandlingCallback.Type type,
+                                          @Nullable final TaskMessageCallback messageCallback) {
+            super(context, type, messageCallback);
+            this.username = username;
+        }
+
+        public AccountDataUpdatedCallback(@NonNull final Context context,
+                                          @NonNull final String username,
+                                          @NonNull final ErrorHandlingCallback.Type type,
+                                          @Nullable final TaskProgressCallback progressCallback,
+                                          @Nullable final TaskMessageCallback messageCallback) {
+            super(context, type, progressCallback, messageCallback);
+            this.username = username;
+        }
 
-    public Account updateAccount(@NonNull String username, @NonNull String field, @Nullable Object value) throws HttpException {
-        final Account updatedAccount = userService.updateAccount(username, Collections.singletonMap(field, value));
-        EventBus.getDefault().post(new AccountDataLoadedEvent(updatedAccount));
-        // Update the logged in user's ProfileImage
-        loginPrefs.setProfileImage(username, updatedAccount.getProfileImage());
-        return updatedAccount;
+        @Override
+        protected void onResponse(@NonNull final Account account) {
+            EventBus.getDefault().post(new AccountDataLoadedEvent(account));
+            // Store the logged in user's ProfileImage
+            loginPrefs.setProfileImage(username, account.getProfileImage());
+        }
     }
 
-    public void setProfileImage(@NonNull String username, @NonNull final File file) throws HttpException, IOException {
+    public Call<ResponseBody> setProfileImage(@NonNull String username, @NonNull final File file) {
         final String mimeType = "image/jpeg";
-        logger.debug("Uploading file of type " + mimeType + " from " + file.toString());
-        userService.setProfileImage(
+        return userService.setProfileImage(
                 username,
                 "attachment;filename=filename." + MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType),
-                new TypedFile(mimeType, file));
-        EventBus.getDefault().post(new ProfilePhotoUpdatedEvent(username, Uri.fromFile(file)));
+                RequestBody.create(MediaType.parse("image/jpeg"), file));
     }
 
-    public void deleteProfileImage(@NonNull String username) throws HttpException {
-        userService.deleteProfileImage(username);
-        EventBus.getDefault().post(new ProfilePhotoUpdatedEvent(username, null));
-        // Delete the logged in user's ProfileImage
-        loginPrefs.setProfileImage(username, null);
-    }
+    public static class ProfileImageUpdatedCallback extends ErrorHandlingCallback<ResponseBody> {
```

Comment:
Yeah, it's something that would be needed once we switch over to using Retrofit's asynchronous API. I have explained the reason for not being able to use it currently at https://github.com/edx/edx-app-android/pull/799#discussion_r77332841.